### PR TITLE
[NMA-645] Private memo UI

### PR DESCRIPTION
**Pull request by Syn-McJ**
The UI + saving memo into the metadata table

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/util/KeyboardUtil.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        } else {
+            root.setPadding(0, 0, 0, defaultPadding)
+        }
+    }
+
+    fun enableAdjustLayout(window: Window, rootView: View) {
+        this.decorView = window.decorView
+        this.rootView = rootView
+        this.defaultPadding = rootView.paddingBottom
+        decorView!!.viewTreeObserver.addOnGlobalLayoutListener(layoutListener)
```
Comment (Syn-McJ) [inline common/src/main/java/org/dash/wallet/common/util/KeyboardUtil.kt:87]: For some reason, `adjustResize` doesn't work on the `MainActivity` (unlike `StakingActivity` where it works fine). Perhaps the problem is in the `OffsetDialog` - in any case, I'm raising the "save" button above the keyboard manually until this is figured out.

Some other keyboard functionality is moved here from the wallet module class.

File: wallet/src/de/schildbach/wallet/ui/TransactionResultViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import org.dash.wallet.common.WalletDataProvider
+import org.dash.wallet.common.data.TaxCategory
+import org.dash.wallet.common.data.TransactionMetadata
+import org.dash.wallet.common.services.TransactionMetadataProvider
+import org.slf4j.LoggerFactory
+import javax.inject.Inject
+
+@FlowPreview
+@HiltViewModel
+class TransactionResultViewModel @Inject constructor(
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/TransactionResultViewModel.kt:37]: Lifted from taxbit-related code

**Review by HashEngineering — APPROVED** [review]
Looks good.

**Comment by HashEngineering** [comment]
Besides the line breaks on one line that causes a merge conflict.

**Review by HashEngineering — APPROVED** [review]

**Pull request merged**